packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -339,9 +339,9 @@ class PopupMenuItemState<T, W extends PopupMenuItem<T>> extends State<W> {
   /// the menu route.
   @protected
   void handleTap() {
-    widget.onTap?.call();
-
     Navigator.pop<T>(context, widget.value);
```

Comment:
Can you add a comment says
// Need to pop the navigator first in case onTap may push new route
// onto navigator.

Reply:
Yes ofc. I will do that tomorrow.